src/node_buffer.cc

```diff
@@ -675,6 +675,48 @@ void Fill(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void StringWriteWithoutBuffer(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = args.GetIsolate();
+
+  if (!args[0]->IsString())
+    return env->ThrowTypeError("Source must be a string");
+  enum encoding source_encoding = ParseEncoding(isolate, args[1], UTF8);
+  enum encoding target_encoding = ParseEncoding(isolate, args[2], UTF8);
+
+  Local<String> str = args[0]->ToString(isolate);
+
+  if (source_encoding == HEX && str->Length() % 2 != 0)
+    return env->ThrowTypeError("Invalid hex string");
+
+  const size_t length = StringBytes::Size(isolate, str, source_encoding);
+  size_t actual = 0;
+  char* data = nullptr;
+
+  if (length > 0) {
+    data = static_cast<char*>(BUFFER_MALLOC(length));
```

Comment:
Off-topic question: I don't understand why `BUFFER_MALLOC` is a macro instead of a function.